src/borg/helpers/time.py

```diff
@@ -110,31 +111,29 @@ def format_timedelta(td):
 
 
 def calculate_relative_offset(format_string, from_date, earlier=False):
-    """ Calculates offset based on a relative marker. 7d (7 days), 8m (8 months)
-        earlier: whether offset should be calculated to an earlier time.
+    """Calculates offset based on a relative marker. 7d (7 days), 8m (8 months)
+    earlier: whether offset should be calculated to an earlier time.
     """
     if from_date is None:
         from_date = archive_ts_now().date()
 
-    if format_string is None:
-        return from_date
-
-    day_offset_regex = re.compile(r'\d+d')
-    month_offset_regex = re.compile(r'\d+m')
+    if format_string is not None:
+        day_offset_regex = re.compile(r"(?P<offset>\d+)d")
+        month_offset_regex = re.compile(r"(?P<offset>\d+)+m")
 
-    day_offset_str = day_offset_regex.search(format_string)
-    if day_offset_str is not None:
-        day_offset = int(day_offset_str.group()[:-1])
-        day_offset *= -1 if earlier else 1
-        return from_date + timedelta(days=day_offset)
+        day_offset_match = day_offset_regex.search(format_string)
+        if day_offset_match:
+            day_offset = int(day_offset_match.group("offset"))
+            day_offset *= -1 if earlier else 1
+            return from_date + timedelta(days=day_offset)
 
-    month_offset_str = month_offset_regex.search(format_string)
-    if month_offset_str is not None:
-        month_offset = int(month_offset_str.group()[:-1])
-        month_offset *= -1 if earlier else 1
-        return offset_n_months(from_date, month_offset)
+        month_offset_match = month_offset_regex.search(format_string)
+        if month_offset_match:
+            month_offset = int(month_offset_match.group("offset"))
+            month_offset *= -1 if earlier else 1
+            return offset_n_months(from_date, month_offset)
```

Comment:
maybe a separate helper function that parses stuff like 42d and returns a timedelta for that would be generally useful (and maybe we could use it even at other places later).

Reply:
Think returning a timedelta might be tricky without a given 'from_date' since timedelta can't account for 'months' independently due to the varying number of days within months. How should I proceed?

Reply:
oh, right. and same issue we have with leap years.